storage_service/locations/models/dspace_rest.py

```
@@ -0,0 +1,379 @@
+
+        dspace_item = response.json()
+
+        LOGGER.info('DSpace item: %s', dspace_item)
+
+        if package.package_type == 'DIP':
+            LOGGER.info('Package is DIP. Splitting package.')
+
+            for root, dirs, files in os.walk(source_path):
+                for name in files:
+                    with open(os.path.join(root, name), 'r') as f:
+                        content = f.read()
+
+                    bitstream_url = self.rest_url + '/items/' + dspace_item['uuid']
+                    bitstream_url += '/bitstreams?name=' + urllib.quote(name)
+
+                    try:
+                        requests.post(bitstream_url,
+                                      headers=headers,
+                                      cookies={'JSESSIONID': dspace_sessionid},
+                                      data=content,
+                                      verify=False)
+                        LOGGER.debug('%s successfully sent to %s', name, bitstream_url)
+                    except Exception:
+                        LOGGER.info('Error sending %s, to %s', name, bitstream_url)
+
+                    # LOGGER.info('Path: %s', root)
+            shutil.rmtree(source_path)
+
+            try:
+                # Create digital object in ArchivesSpace linking to DIP
+                client = archivesspace.ArchivesSpaceClient(self.archivesspace_url, self.archivesspace_user,
+                                                           self.archivesspace_password, self.archivesspace_port,
+                                                           self.archivesspace_repository)
+            except Exception:
+                LOGGER.error('Could not login to ArchivesSpace server: %s, port: %s, user: %s, repository: %s',
+                             self.archivesspace_url,
+                             self.archivesspace_port,
+                             self.archivesspace_user,
+                             self.archivesspace_repository)
```

Again, you should probably raise the caught exception here after logging or raise a custom exception because a `NameError` is just going to be raised in the next statement when the non-existent `client` is used and that may be more confusing I think.

---

Yes. Good point.

---

You can't pass a bunch of string interpolation arguments to an exception constructor like you can when calling a method on a logger. Therefore, your statement:


    raise Exception('Could not login to ArchivesSpace server: %s, port: %s, user: %s, repository: %s',
                self.as_url,
                self.as_url.port,
                self.as_user,
                self.as_repository)

has to be changed to something like:

    raise Exception('Could not login to ArchivesSpace server: {}, port: {}, user: {}, repository: {}'.format(
                self.as_url,
                self.as_url.port,
                self.as_user,
                self.as_repository))